### PR TITLE
Fix apache snippet

### PR DESCRIPTION
**Pull request by fao89**
https://pulp.plan.io/issues/8573
fixes: #8573

### CODE REVIEW

**Review by mdellweg — APPROVED** [review]

**Comment by mdellweg** [comment]
re https://pulp.plan.io/issues/8130

**Comment by fao89** [comment]
> re https://pulp.plan.io/issues/8130

yep, I'm still working on having nginx and apache more similar on pulp_installer, but this change will be needed anyway.

**Comment by ipanova** [comment]
@fao89 with this snippet change the redirect works properly however whenever it gets followed the content is not served and instead a 404 is triggered, should a separate issue be filed to fix this non functional piece? https://pulp.plan.io/issues/8130#note-2

**Comment by fao89** [comment]
> @fao89 with this snippet change the redirect works properly however whenever it gets followed the content is not served and instead a 404 is triggered, should a separate issue be filed to fix this non functional piece? https://pulp.plan.io/issues/8130#note-2

I guess this is some apache confusion between pulp 2 and 3, how can I reproduce the error you pointed?
I think this can be the same issue, probably apache config needs more changes.

Note: could you share your `local.dev-config.yml` ?

**Comment by ipanova** [comment]
> > @fao89 with this snippet change the redirect works properly however whenever it gets followed the content is not served and instead a 404 is triggered, should a separate issue be filed to fix this non functional piece? https://pulp.plan.io/issues/8130#note-2
> 
> I guess this is some apache confusion between pulp 2 and 3, how can I reproduce the error you pointed?
> I think this can be the same issue, probably apache config needs more changes.
> 
> Note: could you share your `local.dev-config.yml` ?

it happens on the `pulp2-nightly-pulp3-source-centos7` box, specifically with container content, rpm content seemed to be served ok.

**Comment by fao89** [comment]
@ipanova I think it fixes the issue, please test this. I only tested for `podman run` scenario, not sure if it breaks other places

**Comment by ipanova** [comment]
> @ipanova I think it fixes the issue, please test this. I only tested for `podman run` scenario, not sure if it breaks other places

will do, thank you!

File: pulp_container/app/webserver_snippets/apache.conf [hunk 1]
```diff
@@ -1,8 +1,8 @@
-ProxyPass /v2 http://${pulp-api}/v2
-ProxyPassReverse /v2 http://${pulp-api}/v2
+ProxyPass /v2 ${pulp-api}/pulp/container
```
Comment (ipanova) [inline pulp_container/app/webserver_snippets/apache.conf:1]: this should stay /v2

File: pulp_container/app/webserver_snippets/apache.conf [hunk 2]
```diff
@@ -1,8 +1,8 @@
...
+ProxyPassReverse /v2 ${pulp-api}/pulp/container
```
Comment (ipanova) [inline pulp_container/app/webserver_snippets/apache.conf:2]: same here

**Comment by ipanova** [comment]
@fao89 This is what made it work for me
```
$ git diff
diff --git a/pulp_container/app/webserver_snippets/apache.conf b/pulp_container/app/webserver_snippets/apache.conf
index c03008c..f52664a 100644
--- a/pulp_container/app/webserver_snippets/apache.conf
+++ b/pulp_container/app/webserver_snippets/apache.conf
@@ -1,8 +1,8 @@
-ProxyPass /v2 http://${pulp-api}/v2
-ProxyPassReverse /v2 http://${pulp-api}/v2
+ProxyPass /v2 ${pulp-api}/v2
+ProxyPassReverse /v2 ${pulp-api}/v2
 
-ProxyPass /pulp/container http://${pulp-content}/v2
-ProxyPassReverse /pulp/container http://${pulp-content}/v2
+ProxyPass /pulp/container ${pulp-content}/pulp/container
+ProxyPassReverse /pulp/container ${pulp-content}/pulp/container
 
-ProxyPass /token http://${pulp-api}/token
-ProxyPassReverse /token http://${pulp-api}/token
+ProxyPass /token ${pulp-api}/token
+ProxyPassReverse /token ${pulp-api}/token

```

**Comment by ipanova** [comment]
To confirm it works
```
$ http pulp2-nightly-pulp3-source-centos7.lan/v2/
HTTP/1.1 200 OK
Access-Control-Expose-Headers: Correlation-ID
Allow: GET, HEAD, OPTIONS
Connection: Keep-Alive
Content-Length: 2
Content-Type: application/json
Correlation-ID: 430d87b5e6244d8e9021f6c5393c2e8a
Date: Wed, 21 Apr 2021 12:58:32 GMT
Docker-Distribution-Api-Version: registry/2.0
Keep-Alive: timeout=5, max=10000
Server: gunicorn/20.0.4
Vary: Accept
X-Frame-Options: SAMEORIGIN

{}


(pulp) [vagrant@pulp2-nightly-pulp3-source-centos7 ~]$ http pulp2-nightly-pulp3-source-centos7.lan/v2/ina/manifests/latest
HTTP/1.1 302 Found
Access-Control-Expose-Headers: Correlation-ID
Allow: GET, PUT, HEAD, OPTIONS
Connection: Keep-Alive
Content-Length: 0
Content-Type: text/html; charset=utf-8
Correlation-ID: 0174974a8f0a4488ada36c9582bfd545
Date: Wed, 21 Apr 2021 12:58:39 GMT
Docker-Distribution-Api-Version: registry/2.0
Keep-Alive: timeout=5, max=10000
Location: http://pulp2-nightly-pulp3-source-centos7.lan/pulp/container/ina/manifests/latest?validate_token=04c802c69533d8d74fdef8715c90718e23beb4349eceb0326c2f2353d6aa4a00:94177afbd5ce9870d5e499b58121a0ecd64e004bfa47656f29e010c19703e5a6
Server: gunicorn/20.0.4
X-Frame-Options: SAMEORIGIN




(pulp) [vagrant@pulp2-nightly-pulp3-source-centos7 ~]$ http HEAD pulp2-nightly-pulp3-source-centos7.lan/v2/ina/manifests/latest --follow
HTTP/1.1 200 OK
Connection: Keep-Alive
Content-Length: 2842
Content-Type: application/vnd.docker.distribution.manifest.v1+prettyjws; charset=utf-8
Date: Wed, 21 Apr 2021 12:58:46 GMT
Docker-Content-Digest: bd35eca337194f65a12f691e88ba980e000f0fabe79f58e6adf5a51ec820cbe2
Docker-Distribution-API-Version: registry/2.0
Keep-Alive: timeout=5, max=9999
Server: Python/3.6 aiohttp/3.7.4


```

File: CHANGES/8573.bugfix [hunk 3]
```diff
@@ -0,0 +1 @@
+Remove scheme from apache snippet
```
Comment (ipanova) [inline CHANGES/8573.bugfix:1]: can this be updated to `fixed apache snippet config and removed sheme`?

**Review by ipanova — APPROVED** [review]
Thank you so much for figuring this out! :octocat:

**Pull request merged**